lib/msf/core/rpc/v10/rpc_session.rb

```diff
@@ -163,6 +163,45 @@ def rpc_shell_upgrade( sid, lhost, lport)
     { "result" => "success" }
   end
 
+  # Writes to an interactive session (i.e. DB and SMB sessions). Note that you will to manually add a newline at the
+  # end of your input so the system will process it.
+  # You may want to use #rpc_interactive_shell_read to retrieve the output.
+  #
+  # @param [Integer] sid Session ID.
+  # @param [String] data The data to write.
+  #
+  # @return [Hash] A hash indicating the action was successful. It contains the following key:
+  # * 'result' [String] A message that says 'success'
+  #
+  # @example Here's how you would use this from the client:
+  # rpc.call('session.interactive_shell_write', 2, "DATA")
+  # => {"result"=>"success"}
+  def rpc_interactive_shell_write(sid, data)
```

Comment:
Do we need a corresponding `def rpc_meterpreter_run_single( sid, data)` call

https://github.com/rapid7/metasploit-framework/blob/0399846c6de104054957b4f0f3087b8cec267581/lib/msf/core/rpc/v10/rpc_session.rb#L400

Reply:
I need help understanding how this is connected

Reply:
Understood, yes, that makes sense

Reply:
Will put that on my notes and make a ticket later, won't do in this PR. I think if we want to go deeper on API changes it'll need more thought process.